arguments.py

```diff
@@ -2,33 +2,67 @@
 import argparse
 import sys
 import logging
+from argcomplete.completers import EnvironCompleter, ChoicesCompleter
+import argcomplete
 from logger import logger, configure_logger
 
+VALID_STEPS = ["pre","masters","workers","post"]
+
+
+def yaml_completer(prefix, parsed_args, **kwargs):
+    return (f for f in os.listdir('.') if (f.endswith(('.yaml', '.yml')) and f.startswith(prefix)))
+
+
+def step_completer(prefix, parsed_args, **kwargs):
+    if not prefix:
+        return VALID_STEPS
+
+    steps_entered = prefix.split(',')
+
+    available_steps = list(set(VALID_STEPS) - set(steps_entered))
```

Comment:
why `list(...)` ?

Reply:
I agree the list is unnecessary.
Originally I made it a list because I thought iterating over a set was unordered/random. But since the list will be made by iterating over the set, the list does not have a particular order either.